src/server/replica.cc

```diff
@@ -160,7 +164,16 @@ void Replica::Stop() {
 
   // Make sure the replica fully stopped and did all cleanup,
   // so we can freely release resources (connections).
-  sync_fb_.join();
+
+  // TODO: Vlad, shouldnt we call here Replica::JoinAllFlows()?
+  // can it be that sync_fb_ of execution_fb_ are valid? Stop is called from server family
+  // not for replica flows
+  if (sync_fb_.joinable()) {
+    sync_fb_.join();
+  }
+  if (execution_fb_.joinable()) {
+    execution_fb_.join();
+  }
 }
```

Comment:
The logic is as follows (and its kind of bad the replica and sub-replica are on class - but thats what we have for now).

1. We cancel the context
2. It unblocks
3. The current task (i.e. full/stable sync) must do cleanup when it exists and returns an error - i.e. join the flows
4. We wait for sync_fb_ to join on the main replica class - so when the loop exited, the task must have done cleanup

We can join all flows - but that shouldn't be needed here - it happens either way. Besides, we can only join if the flows are not in a transitioning state, so we have to lock by the transition mutex... so it makes it all more complicated

Reply:
I see now that in the main replica sync_fb_ is set to MainReplicationFb. I thought that it is only used in the sub-replica instances and this is why I thought that there was a bug here that you wanted to join the sub flows here. Never mind.
I will remove the execution_fb_.join() from here

Reply:
You can leave a comment there, that sync_fb_ is the MainReplicationFb and is responsible for stopping everything